src/index.tsx

```diff
@@ -11,117 +11,160 @@ import Animated, {
   useSharedValue,
 } from 'react-native-reanimated';
 
+type MarqueeDirection = 'up' | 'left' | 'right' | 'down';
+
 const AnimatedChild = ({
   index,
   children,
   anim,
+  textHeight,
   textWidth,
   spacing,
+  direction = 'right',
 }: React.PropsWithChildren<{
   index: number;
   anim: SharedValue<number>;
+  textHeight: SharedValue<number>;
   textWidth: SharedValue<number>;
   spacing: number;
+  direction?: MarqueeDirection;
 }>) => {
   const stylez = useAnimatedStyle(() => {
+    const vertical = direction === 'up' || direction === 'down';
```

Comment:
Since this is a boolean value, I will rename it to `isVertical` and `isHorizontal`.